lib/service/storage/elasticsearch.js

```diff
@@ -2711,6 +2721,22 @@ class ElasticSearch extends Service {
     return searchBody;
   }
 
+  /**
+   * Throw if any script keyword is contained in the object  
+   * 
+   * @param {Object} object
+   */
+  _scriptCheck(object) {
+    for (const [key, value] of Object.entries(object)) {
+      if (this.scriptKeys.includes(key)) {
+        throw kerror.get('invalid_query_keyword', key);
+      }
+      else if (typeof value === 'object' && value !== null) {
```

Comment:
`isPlainObject` should be used instead: as it is, this function will also scan arrays. This won't have any impact on the feature (array indexes are numerals), but this makes Kuzzle do useless work.

Reply:
Indeed we should avoid useless work but we also need to go deeper in the array (in a boolean query for example), I'll correct that consequently.

Reply:
oh, you're right, I didn't consider arrays containing subqueries

don't change it then, it's fine like this (maybe add a quick comment to prevent others to make the same mistake than me?)